### ISSUE TITLE
Support `.clear()` on date inputs

### ISSUE THREAD

**Issue by aciccarello** [cypress-io/cypress#2738]
Using `.clear()` on a date field should clear the value rather than causing an error. This may affect other field types.

[Related StackOverflow question](https://stackoverflow.com/q/53196292/4252741)

### Current behavior:

The [`.clear()`](https://docs.cypress.io/api/commands/clear.html#Syntax) command is an alias for `.type({selectall}{backspace})`. However, typing in date fields is limited to strings matching the date format. This causes Cypress to throw an error.

> CypressError: Typing into a date input with cy.type() requires a valid date with the format 'yyyy-MM-dd'. You passed: {selectall}{del}

### Desired behavior:

The type command should allow `{selectall}{del}` for date inputs and clear the input value.

### Versions
Tested on Cypress 3.1.0

**Comment by jennifer-shehane** [comment]
Have confirmed, this does error when trying to clear a date field and it should not error. 

```js
    cy.get('input').type('2019-08-07')
    cy.get('input').clear()
```

**Comment by lilaconlee** [comment]
This has been fixed and we'll update the issue and the [changelog](https://docs.cypress.io/guides/references/changelog.html) when it's released.

**Comment by aciccarello** [comment]
@lilaconlee Look like this was released in 3.1.5 however I don't see that noted in the changelog.

**Comment by jennifer-shehane** [comment]
Thanks @aciccarello, this was released in [3.1.5](https://on.cypress.io/changelog#3-1-5). The changelog has been updated to reflect this.
